src/internal/ajax/ajax.ts

```diff
@@ -507,3 +517,14 @@ function isURLSearchParams(body: any): body is URLSearchParams {
 function isReadableStream(body: any): body is ReadableStream {
   return typeof ReadableStream !== 'undefined' && body instanceof ReadableStream;
 }
+
+function readCookie(name: string): string {
+  try {
+    return document?.cookie.match(new RegExp(`(^|;\\s*)(${name})=([^;]*)`))?.pop() ?? '';
+  } catch (err) {
+    if (err instanceof ReferenceError) {
```

Comment:
Further explanation of the in-lined stuff above. The `try/catch` isn't necessary because it's really just here to catch the `ReferenceError` and then rethrow. If we check `cookie` inline like `document?.cookie?.match`, then there's no possibility of a `ReferenceError` and the `try/catch` has no purpose (because we're just rethrowing).

Reply:
Optional chaining defends against `TypeError`s that get thrown due to accessing properties of `null` or `undefined`.

In Node.js, the `document` variable is not even initialized, so accessing the variable `document` itself (even without accessing properties on that object!) throws a `ReferenceError` that optional chaining will not prevent.

That being said, I have thought about this some and decided that I am probably being overly-defensive. I think that it is reasonable to leave it to the user to only use `xsrf*Name` options in the DOM, and to allow the `ReferenceError` to throw if they try to use it in Node. If someone _really_ wants it in Node then the library can always change later.

Reply:
I see what you're getting at. Thank you for the explanation. I'll try to revisit this in another pass. I suspect there are some places where we need to just do a sanity check on references for environments like node and older runtimes. And I also suspect that there's a compact way to do this.